include/xgboost/data.h

```diff
@@ -287,10 +289,23 @@ class SortedCSCPage : public SparsePage {
   explicit SortedCSCPage(SparsePage page) : SparsePage(std::move(page)) {}
 };
 
+class EllpackPageImpl;
```

Comment:
Can you explain briefly why this class needs [pimpl](https://en.cppreference.com/w/cpp/language/pimpl)?

Reply:
Mainly to get CUDA-specific implementation details out of the header, since DMatrix is used all over the place in CPU-only code.